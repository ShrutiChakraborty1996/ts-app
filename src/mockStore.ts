import thunk from "redux-thunk";
import configureStore, {MockStore} from "redux-mock-store";
import TodoType from "./types/TodoType";

export const generateMockStore = (state: TodoType[] | {}): MockStore<TodoType[] | {}> => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(state);
    store.clearActions();
    return store;
};