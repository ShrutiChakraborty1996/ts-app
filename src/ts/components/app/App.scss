@import '../../../assets/stylesheets/mixin';

.App {
  text-align: center;

  @include themify($themes){
    color: themed('color');
    background-color: themed('bg');
  }

  &-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
  }

  &-header {
    height: 150px;
    padding: 20px;

  }

  &-title {
    font-size: 1.5em;
  }

  &-intro {
    font-size: large;
    padding: 10px;
  }
}


@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
