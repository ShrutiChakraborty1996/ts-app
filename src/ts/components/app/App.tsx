import * as React from 'react';
import './App.scss';

import logo from '../../../assets/images/logo.svg';
import Todos from "../../../ts/containers/todos/Todos";
import Theme from "../../../ts/containers/themes/ThemeContainer";
import {connect} from "react-redux";

class App extends React.Component<any, any> {
    constructor(props: any){
        super(props);
    }

    public render() {
        return (
            <div className={`App theme-${this.props.selectedTheme}`}>
                <header className="App App-header">
                    <img src={logo} className="App App-logo" alt="logo"/>
                    <h1 className="App App-title">Welcome to React</h1>
                </header>
                <Theme/>
                <Todos/>
            </div>

        );
    }
}

const mapStateToProps = (state: any) => {
    return {
        selectedTheme: state.theme.selectedTheme
    }
};

export default connect(mapStateToProps,undefined)(App);
