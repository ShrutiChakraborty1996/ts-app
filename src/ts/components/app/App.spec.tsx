import * as React from 'react';
import App from './App';
import {generateMockStore} from "../../mockStore";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";

describe('App component', () => {
    const initialState: any = [];
    let mockStore = generateMockStore(initialState);

    beforeEach(() => {
        mockStore.clearActions();
    });

    describe('when app is passed', () => {
        const mockState = {
            theme: {selectedTheme: "sgp"},
            todos: [
                {title: "sample"}
            ]
        };
        mockStore = generateMockStore(mockState);

        it('should render App Component', () => {
            const component = renderer.create(<Provider store={mockStore}><App/></Provider>);
            expect(component).toMatchSnapshot();
        });
    });
});
