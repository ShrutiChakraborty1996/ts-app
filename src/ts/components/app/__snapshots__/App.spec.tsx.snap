// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component when app is passed should render App Component 1`] = `
<div
  className="App theme-sgp"
>
  <header
    className="App App-header"
  >
    <img
      alt="logo"
      className="App App-logo"
      src={undefined}
    />
    <h1
      className="App App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="themes"
  >
    <form
      onChange={[Function]}
    >
      <select
        id="theme"
        name="themes"
      >
        <option
          value="dark"
        >
          Singapore
        </option>
        <option
          value="light"
        >
          India
        </option>
      </select>
    </form>
  </div>
  <div
    className="todo-form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
      />
      <button
        type="submit"
      >
        +
      </button>
    </form>
    <ul>
      <li>
        sample
      </li>
    </ul>
  </div>
  <form
    onClick={[Function]}
  >
    <div>
      <input
        defaultChecked={true}
        name="set1"
        type="radio"
        value="something 1"
      />
      <label>
        something 1 
      </label>
    </div>
    <div>
      <input
        name="set1"
        type="radio"
        value="something 2"
      />
      <label>
        something 2 
      </label>
    </div>
    <div>
      <input
        name="set1"
        type="radio"
        value="something 3"
      />
      <label>
        something 3 
      </label>
    </div>
  </form>
</div>
`;
